src/containers/record-modal.jsx

```diff
@@ -0,0 +1,123 @@
+const bindAll = require('lodash.bindall');
+const PropTypes = require('prop-types');
+const React = require('react');
+const VM = require('scratch-vm');
+const WavEncoder = require('wav-encoder');
+const {connect} = require('react-redux');
+
+const RecordModalComponent = require('../components/record-modal/record-modal.jsx');
+
+const {
+    closeSoundRecorder
+} = require('../reducers/modals');
+
+class RecordModal extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'handleRecord',
+            'handleStopRecording',
+            'handlePlay',
+            'handleStopPlaying',
+            'handleBack',
+            'handleSubmit',
+            'handleCancel'
+        ]);
+
+        this.state = {
+            samples: null,
+            encoding: false,
+            levels: null,
+            playing: false,
+            recording: false,
+            sampleRate: null
+        };
+    }
+    handleRecord () {
+        this.setState({recording: true});
+    }
+    handleStopRecording (samples, sampleRate, levels) {
+        this.setState({samples, sampleRate, levels, recording: false});
+    }
+    handlePlay () {
+        this.setState({playing: true});
+    }
+    handleStopPlaying () {
+        this.setState({playing: false});
+    }
+    handleBack () {
+        this.setState({playing: false, samples: null});
+    }
+    handleSubmit () {
+        this.setState({encoding: true});
+
+        // Allow enough time for UI refresh before starting encoding
```

Comment:
So my hope was that if the wav encoding took a bit of time, we would have enough time to switch this to say `Loading...` before starting the encoding process. Does putting this in a callback of `setState` do that?